cellmlmanip/singularity_fixes.py

```diff
@@ -0,0 +1,219 @@
+from math import isclose
+
+import networkx as nx
+from sympy import (
```

Comment:
Should this whole module be private?

Single cellmlmanip is a library, we need to carefully think about its API.

This code adds a public module `cellmlmanip.singularity_fixes` that has public classes `Abs`, `Add`, `FLOAT_PRECISION`, `ONE`, etc.
So if we're saying this module is public, then these should probably not be imported with a `from`, but just accessed via `sympy.Abs` etc.